src/main/scala/com/vngrs/etl/Extractor.scala

```diff
@@ -0,0 +1,27 @@
+package com.vngrs.etl
+
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.Row
+
+import scala.reflect.ClassTag
+
+trait Extractor[A] {
+  protected[etl] def apply(sc: SparkContext): RDD[A]
```

Comment:
@msimav and @kubudi, I am actually concerned about protecting `apply` method within `etl` scope. Is it going to let users to override `apply` method for custom `Extractor`s?